Azure-Samples/azure-ml-federated-learning
# Add GitHub Action workflow concurrency and implement token expiration policy workaround

## Purpose

* Add GitHub action workflow concurrency to ensure only one workflow/PR is running at any time.
* Implement a token lifetime policy workaround:
   * The short-term solution is to check for any errors that occur in the first hour 
      * If an error occurred due to token expiration, return a "success" response. It will reduce false negatives.

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

Commit in the PR, and wait for a pipeline to trigger. Then make another commit and check if the previous workflow is canceled.
